lib/settings.php

```diff
@@ -11,8 +11,29 @@
 // Call the REDCap Connect file in the main "redcap" directory; enforces permissions.
 require_once(dirname(realpath(__FILE__)) . '/../../../redcap_connect.php');
 require_once(dirname(realpath(__FILE__)) . '/ReportConfig.php');
+require_once(dirname(realpath(__FILE__)) . '/ReportConfigProcessor.php');
 
-$reportConfig = new ReportConfig($project_id, $module);
-$settings = $reportConfig->getReportConfig();
+if (isset($_GET['action'])) {
+    if ($_GET['action'] === 'saveReportSummary') {
+        $requestBody = trim(file_get_contents('php://input'));
 
-print($settings);
+        if (!empty($requestBody)) {
+            $data = json_decode($requestBody, true);
+            $reportSummary = json_decode($data['reportSummary'], true);
+
+            $reportConfig = new ReportConfig($project_id, $module);
+            $reportConfig->saveReportSummary($reportSummary['reportSummary']);
```

Comment:
You may want to add some validation of structure of the summary to ensure it is actually what we believe it is.